### PR TITLE
Improve doc coverage (#44)

### PR DESCRIPTION
**Pull request by yever**
I added documentation to most public elements in `sophia/`, using
the method described in the issue (#44) of adding `#![deny(missing_docs)]`
to `sophia/src/lib.rs`.

~2 doc comments are missing:~
- ~[sophia_api::make_scoped_triple_streaming_mode!(ScopedRioSourceTriple, RioSourceTriple);](https://github.com/pchampin/sophia_rs/blob/master/sophia/src/parser/rio_common.rs#L69)~
- ~[sophia_api::make_scoped_quad_streaming_mode!(ScopedRioSourceQuad, RioSourceQuad);](https://github.com/pchampin/sophia_rs/blob/master/sophia/src/parser/rio_common.rs#L108)~

~I don't know how to add comments to the macro-created types.~

I found out that it's possible to add doc comments to macros (https://stackoverflow.com/a/33999625/4696215) and updated the relevant macros accordingly in a separate commit.

### CODE REVIEW

**Comment by yever** [comment]
I have added the last missing doc comments in the `sophia` crate and now the travis build is green :)

This pull request is now ready to be reviewed. @pchampin

File: sophia/src/parser/trig.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ use std::io::BufRead;
 /// TriG parser based on RIO.
 #[derive(Clone, Debug, Default)]
 pub struct TriGParser {
+    /// The current base IRI.
```
Comment (pchampin) [inline sophia/src/parser/trig.rs:11]:
```suggestion
    /// The base IRI used by this parser to resolve relative IRI-references.
```

File: sophia/src/parser/turtle.rs [hunk 2]
```diff
@@ -11,6 +11,7 @@ use thiserror::Error;
 /// Turtle parser based on RIO.
 #[derive(Clone, Debug, Default)]
 pub struct TurtleParser {
+    /// The current base IRI.
```
Comment (pchampin) [inline sophia/src/parser/turtle.rs:14]:
```suggestion
    /// The base IRI used by this parser to resolve relative IRI-references.
```

File: sophia/src/parser/turtle.rs [hunk 3]
```diff
@@ -25,6 +26,10 @@ impl<B: BufRead> TripleParser<B> for TurtleParser {
     }
 }
 
+/// A wrapper around [`rio_turtle::TurtleError`] the implements [`WithLocation`].
```
Comment (pchampin) [inline sophia/src/parser/turtle.rs:29]:
```suggestion
/// A wrapper around [`rio_turtle::TurtleError`] that implements [`WithLocation`].
```

File: sophia/src/parser/xml.rs [hunk 4]
```diff
@@ -12,6 +12,7 @@ use crate::parser::TripleParser;
 /// N-Triples parser based on RIO.
 #[derive(Clone, Debug, Default)]
 pub struct RdfXmlParser {
+    /// The base IRI.
```
Comment (pchampin) [inline sophia/src/parser/xml.rs:15]:
```suggestion
    /// The base IRI used by this parser to resolve relative IRI-references.
```

File: sophia/src/parser/xml_legacy.rs [hunk 5]
```diff
@@ -99,6 +99,7 @@ pub struct RdfXmlParser {
 }
 
 impl RdfXmlParser {
+    /// Build a new RDF XML parser with the given config.
```
Comment (pchampin) [inline sophia/src/parser/xml_legacy.rs:102]:
```suggestion
    /// Build a new RDF/XML parser with the given config.
```

**Review by pchampin — CHANGES_REQUESTED** [review]
Apart from a few things that I would rather rephrase (see comments), this is great. Thanks a lot.

**Comment by pchampin** [comment]
Finally found a moment to merge this! Thanks again @yever

**Pull request merged**